src/test/rgw/test_redis_lock.cc

```diff
@@ -0,0 +1,59 @@
+#include <gtest/gtest.h>
+#include <rgw_redis_lock.h>
+
+#include <boost/asio/detached.hpp>
+#include <boost/asio/io_context.hpp>
+#include <boost/redis/connection.hpp>
+
+using boost::redis::config;
+using boost::redis::connection;
+
+void rethrow(std::exception_ptr eptr) {
+  if (eptr) std::rethrow_exception(eptr);
+}
+
+class RGWRedisLockTest : public ::testing::Test {
+ protected:
+  boost::asio::io_context io;
+  connection *conn;
+
+  void SetUp() {
+    // Creating the default config
+    config cfg;
+
+    conn = new connection(io);
```

Comment:
you can probably create the shared pointer here, just call `std::make_shared()`.
and use it in the different tests